roxie/ccd/ccdactivities.cpp

```diff
@@ -567,6 +567,7 @@ class CRoxieAgentActivity : implements CInterfaceOf<IRoxieAgentActivity>, implem
     // Not yet thought about these....
 
     virtual char *getWuid() { throwUnexpected(); } // caller frees return string.
+    virtual unsigned getWorkflowId() const override { throwUnexpected(); }
```

Comment:
trivial: in fact expected, but unimplemented at the moment? (maybe change to UNIMPLEMENTED_X("getWorkflowId"); )

Reply:
This isn't missing functionality.  This getWorkflowId shouldn't be called.  I think it should remain as throwUnexpected.

Reply:
Since it's not used anyway yet, it doesn't matter anyway,
but do still wonder if could be via CRoxieAgentActivity::onCreate()
```
        basehelper->onCreate(this, NULL, &serializedCreate);
```

where this is the ICodeContext of CRoxieAgentActivity, where this is the implemented member.

Anyway, if it is an issue it can be revisited in https://track.hpccsystems.com/browse/HPCC-31154.